client/state/shortcodes/reducer.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import { LOAD_STATUS } from './constants';
+import { shortcodesSchema } from './schema';
+import {
+	SHORTCODE_FETCH,
+	SHORTCODE_RECEIVE
+} from 'state/action-types';
+
+const siteShortcode = ( state = {}, { type, error, data } ) => {
```

Comment:
Shortcode results can vary by site, so we'd need state to reflect this.

Also, rather than track "shortcode" and "shortcodes" separately, you might consider a single `items` object, which maps a site ID, shortcode pair to the received result.

It's fairly common in state to split out the request progress to its own reducer key. See [the page templates reducer](https://github.com/Automattic/wp-calypso/blob/e6eb37a38c8f5041e3cdf285bea03afc8ef989fc/client/state/page-templates/reducer.js#L18-L36) as an example. This could help eliminate the need for a constant and error handling within this reducer. To handle errors, you could create yet another reducer for tracking the error results of a shortcode request.